content/blaze.md

```diff
@@ -0,0 +1,543 @@
+
+Notice as well in this case that we access the global client-side state store `FlowRouter` in the `onCreated()` callback also, storing its value locally in a local `state` dictionary.
+
+XXX: should instead we just read from flow router a second time here: https://github.com/meteor/todos/blob/master/packages/lists-show/lists-show-page.js#L28 ??
+
+<h3 id="fetch-in-smart-components">Fetch in helpers</h3>
+Typically, as [outlined in the ui/ux article](ui-ux.md#smart-components) you should fetch data in the same component that you subscribe in. In Blaze, it's usually simplest to fetch the data in a helper, which you can then use to pass data into a reusable child component. For example, in the `listShowPage`: 
+
+```html
+{{> listsShow todosReady=Template.subscriptionsReady
+  list=(getFullList listIdOnly) todos=listIdOnly.todos}}
+```
+
+<h2 id="reusing-code">Reusing code in Blaze</h2>
+It's not uncommon to want to reuse code between two otherwise unrelated components. There are two principal ways to do this in Blaze.
+
+<h3 id="composition">Composition</h3>
+If possible, it's usually best to try and abstract out the reusable part of the two components that need to share functionality. If you are used to following patterns to [create reusable components](#reusable-components), then it should be simple to reuse those components in many places. 
+
+<h3 id="utility-libraries">Utility libraries</h3>
+It's usually best to keep your view layer as "skinny" as possible and contain a component to whatever specific task it specifically needs to do. If there's heavy lifting involved (such as complicated rendering logic), it often makes sense to abstract it out into a utility library that simply deals with the logic alone and doesn't deal with the Blaze system at all. 
+
+For instance, if a component requires a lot of complicated [D3](d3js.org) code for drawing graphs, it's likely that that code itself could live in a utility library that's called by the component. That makes it easier to abstract the code later and share it between various components that need to all draw graphs.
+
+<h3 id="global-helpers">Global Helpers</h3>
+
+One type of library that is useful is a global Spacebars helper. You can define these with the `Template.registerHelper()` function. Typically you register helpers to do simple things (like rendering dates in a given format) which don't justify a separate sub-component. For instance, you could do:
+
+```js
+Template.registerHelper('shortDate', (date) => {
+  return moment(date).format("MMM do YY");
+});
+```
+
+```html
+<template name="myBike">
+  <dl>
+   <dt>Date registered</dt>
+   <dd>{{shortDate bike.registeredAt}}</dd>
+ </dl>
+</template>
+```
+
+<h2 id="understanding-blaze">Understanding Blaze</h2>
+Although Blaze is a very intuitive rendering system, it does have some quirks and complexities that are worth knowing about when you are trying to do complex things.
+
+<h3 id="re-rendering">Re-rendering</h3>
```

Comment:
Somewhere near the top of this we should mention that the intention is that helpers should be cheap to re-run